lib/virt_autotest/utils.pm

```diff
@@ -1270,6 +1270,10 @@ sub get_guest_regcode {
     my $count = ($args{separator} eq '|' ? scalar(split("\\$args{separator}", $guest)) : scalar(split("$args{separator}", $guest)));
     $regcode = join("$args{separator}", (get_var("SCC_REGCODE", "")) x $count) if (!$regcode);
     $regcode_ltss = join("$args{separator}", (get_var("SCC_REGCODE_LTSS_15", "")) x $count) if (!$regcode_ltss);
+    if (check_var('VMVERSION', '12-SP5')) {
+        my $vmversion = get_var("VMVERSION") =~ s/-/_/r;
+        $regcode_ltss = get_var("SCC_REGCODE_LTSS_$vmversion");
+    }
```

Comment:
This block does not conform to the original way in which ltss code is retrieved.  I think you should continue to used regcode_ltss like this:

```
if 15
$regcode_ltss=ltss_code_15
elsif 12
$regcode_ltss=ltss_code_12
```

There is no need to use VMVERSION.

You can now guest version from guest list.

Reply:
When you add new code to support new functionality I think you should make the overall structure still clean and intact instead of just adding another piece of standalone code. Although we are QA, we still develop code for QA.

Reply:
Agree, see my similar comment https://github.com/SUSE/qa-automation/pull/844#issuecomment-2445777770

Reply:
thanks, I will get guest from parameter: GUEST_LIST

Reply:
```
if 15
$regcode_ltss=ltss_code_15
elsif 12
$regcode_ltss=ltss_code_12
```
This statement is added to support both `SCC_REGCODE_LTSS_15` and `SCC_REGCODE_LTSS_12`? but is the ltss code identical for different arches?

Reply:
oh, I got the points. the arm tests get SCC_REGCODE settings from its own "medium types" on OSD. It makes sence.  

On x86 tests, I don't see `SCC_REGCODE_LTSS_12`. Anyway, you can add this setting in arm test suites.